README.md

```diff
@@ -4,13 +4,36 @@
 - List 3 reasons why asymptotic analysis may be misleading with respect to
   actual performance in practice.
 
+  1) Asymptotic analysis normally only accounts for the average or worst case scenarios in algorithms, but in the real world these cases are often different and unpredictable.
+     
+  2) In the real world, the efficiency of these algorithms can be altered by uncontrollable factors such as the hardware or programming language the algorithm is being ran on.
+     
+  3) In some cases, algorithms run at different speeds depending on what data structure it is working with and this is sometimes unaccounted for.
+  
+
 - Suppose finding a particular element in a binary search tree with 1,000
   elements takes 5 seconds. Given what you know about the asymptotic complexity
   of search in a binary search tree, how long would you guess finding the same
   element in a search tree with 10,000 elements takes? Explain your reasoning.
+  
+  We consider the time complexity of a binary tree to be O(log(n)).
+  If 5 seconds = O(log(1000)),
+     x seconds = O(log(10000)).
+     log(1000) / log(2) is roughly equal to 10.
+     log(10000) / log(2) is roughly equal to 13.3.
+     (5 seconds / 10) = (x seconds / 13.3)
+     It would likely take about 6.5 seconds to find the element in the tree.
+  
 
 - You measure the time with 10,000 elements and it takes 100 seconds! List 3
   reasons why this could be the case, given that reasoning with the asymptotic
   complexity suggests a different time.
 
+  1) There is a possibility that the tree is very poorly balanced, causing the search time to be way higher than it should be.
+ 
+  2) The hardware the program is being run on could be slow.
```

Comment:
But that would affect the smaller input size as well.

Reply:
Slow hardware would affect both a small and large input size, but I think that it would have a more noticeable impact on a larger input size because it has to search the tree for more elements.

Reply:
...but it sounds like this would only be a constant factor?

Reply:
I don't think that slower hardware would be a constant factor because as the size of the tree increases, the limitations of the hardware elements such as the CPU or memory become more obvious when compared to a smaller tree. The smaller tree would be affected but not as significantly.

Reply:
Hmm, but e.g. a frequency difference would be a constant factor. What could cause it to be not a constant factor?

Reply:
There are a few things to consider that could cause it to not be a constant factor. First, the CPUs running the program could be throttling their frequency differently to control their temperatures to prevent overheating. Also, the different CPUs could have different power management and battery saving features that could cause it to run at different speeds.

Reply:
Again it sounds like this would affect the smaller input size as well.

Reply:
Assuming this is a true real-world situation, it is possible that while the program was being run on the smaller input size, the CPU was under low stress and relatively colder. While the program was being run on the larger input size, the CPU could have been under higher stress, possibly from running other programs in the background that it didn't have to deal with while running the program with the smaller input size. If this were the case, the runtime for the smaller input size would have been affected differently than the larger input size.

Reply:
Ok, but this now sounds like you're saying that the environment was different for the different input sizes.

Reply:
Well... I haven't told you anything about how this was run, have I? In particular, I didn't say that everything is run in the same way.